### PR TITLE
Create booking

### PR DESCRIPTION
**Pull request by ZaneMeroff**
### What's this PR do?
Build functionality to the search for available bookings feature for customer
### Where should the reviewer start?
Log in as customer. You can use (customer1 - customer50), at this time the app does not require a password.
### How should this be manually tested?
After logging in, try making a search for available rooms. Live search results can only be seen from using a console.log(signedInUser.availableRooms).
### Potential changes?
Need to account for days search results from days in the past as well as filtering by type of room. Need to update search results on DOM.

@pJanks
@khalidwilliams
@hannahhch

### CODE REVIEW

File: src/domUpdates.js [hunk 1]
```diff
@@ -18,12 +19,16 @@ const domUpdates = {
...
-     hideOrShowElement('show', '.available-res-container');
-     hideOrShowElement('hide', '.book-a-cabin-container');
-     hideOrShowElement('hide', '.past-future-container');
-     hideOrShowElement('show', '.new-search-button');
-  }),
+  displayCustomerSearchResults: () => {
+    let user = giveUser();
+    user.searchDate = $('.date-input').val();
+    user.addAvailableCabinsToBookingSearch()
+    console.log(user.availableRooms);
```
Comment (pJanks) [inline src/domUpdates.js]: Don't forget to remove logs before submitting for review.

File: src/index.js [hunk 2]
```diff
@@ -71,16 +73,26 @@ const checkSignInStatus = () => {
...
     if (user.id === id) {
       return user;
     }
   })
-  signedInUser = new Customer(selectedUser.id, selectedUser.name, allData, getTodaysDate())
+  signedInUser = new Customer(selectedUser.id, selectedUser.name, getTodaysDate())
   signedInUser.findAllBookings()
 }
 
+export const instanciateHotel = () => {
```
Comment (pJanks) [inline src/index.js]: You misspelled instantiate.

File: src/domUpdates.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import $ from 'jquery';
 import Customer from './customer.js'
 import {hideOrShowElement} from './index.js';
+import {giveUser} from './index.js';
```
Comment (pJanks) [inline src/domUpdates.js]: I may be wrong, but I don't think you need the curly braces here or on line 3.

File: src/index.html [hunk 4]
```diff
@@ -25,20 +25,20 @@ <h2 class="h2-heading">a JW Marriott Product</h2>
...
       </div>
 
       <div class="book-a-cabin-container hidden">
         <div class="past-future-heading">
           <h2>book a new reservation</h2>
         </div>
         <div class="select-dates-container">
-          <select class="possible-dates-dropdown dates">
+          <input type="text" class="date-input" placeholder="year/month/day">
+          <!-- <select class="possible-dates-dropdown dates">
```
Comment (pJanks) [inline src/index.html]: Consider removing commented out code if you have already solved the problem you were using it for in the first place.

File: src/hotel.js [hunk 5]
```diff
@@ -1,14 +1,17 @@
...
-  constructor(users, bookings, rooms) {
-    this.users = users;
-    this.bookings = bookings;
+  constructor(users, rooms, bookings, date) {
+    this.allUsers = users;
     this.rooms = rooms;
-    // this.today = this.getTodaysDate();
-    this.currentUser;
-    this.date;
+    this.bookings = bookings;
```
Comment (pJanks) [inline src/hotel.js]: Good job building out a class to hold all of the data you are fetching.

File: src/domUpdates.js (hunk 1 above)
Comment (ZaneMeroff) [reply src/domUpdates.js]: Looks like I missed that one. Thanks for looking out! I'll make sure to look for those for my next PR.

File: src/index.js (hunk 2 above)
Comment (ZaneMeroff) [reply src/index.js]: Good eye @pJanks! I'll correct that immediately.

File: src/domUpdates.js (hunk 3 above)
Comment (ZaneMeroff) [reply src/domUpdates.js]: Not sure about that one. I will look into it. Thanks for brining it to my attention.

File: src/index.html (hunk 4 above)
Comment (ZaneMeroff) [reply src/index.html]: Thanks for finding this. Yes, that comment is no longer relevant. I will remove it.

File: src/hotel.js (hunk 5 above)
Comment (ZaneMeroff) [reply src/hotel.js]: Thank! After giving it some thought I decided all rooms, users, and bookings should live on the hotel class. I was also debating if the customer class should hold onto the current date, or should it be kept in the hotel class. @pJanks What do you think?

**Pull request merged**